src/_P088_HeatpumpIR.ino

```diff
@@ -205,42 +167,38 @@ boolean Plugin_088(byte function, struct EventStruct *event, String& string)
           sendMinute = minute();
           sendWeekday = weekday();
 #endif
-          int i = 0;
-          do
-          {
-            const char* shortName = heatpumpIR[i]->model();
+          heatpumpIR = HeatpumpIRFactory::create(heatpumpModel.c_str());
 
-            if (strcmp_P(heatpumpModel.c_str(), shortName) == 0)
-            {
-              enableIR_RX(false);
-              heatpumpIR[i]->send(*Plugin_088_irSender, powerMode, operatingMode, fanSpeed, temperature, vDir, hDir);
-              enableIR_RX(true);
-              addLog(LOG_LEVEL_INFO, F("P088: Heatpump IR code transmitted"));
+          if (heatpumpIR != NULL) {
+            enableIR_RX(false);
+            heatpumpIR->send(*Plugin_088_irSender, powerMode, operatingMode, fanSpeed, temperature, vDir, hDir);
+            enableIR_RX(true);
+
+            delete heatpumpIR;
```

Comment:
You do create/delete it every run. It is for sure a bit more resource efficient, but please test it for a while.
You may run into allocation issues later on if the heap gets fragmented, so it may not be able to allocate larger blocks of data.

Do you have an rough estimate on how big the objects are you like to allocate?

Reply:
The amount of data per class instance should be quite minimal. It only has two static const class variables, and a couple of methods.

Reply:
OK, then the chance of allocation issues due to heap fragmentation is minimal or almost non existent.